### PR TITLE
Salesforce Composite 2.2.0: connector reference

### PR DESCRIPTION
**Pull request by dancazacu1**

### CODE REVIEW

**Review by george-hoffer — APPROVED** [review]

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 1]
```diff
@@ -1,889 +1,1187 @@
...
+| Consumer Secret a| String |  Your app's client secret (consumer secret in remote access detail). |  | x
+| Username a| String |  Username used to initialize the session. |  | x
+| Password a| String |  Password used to authenticate the user. |  | x
+| Security Token a| String |  User's security token, which can be omitted if your IP has been white-listed on Salesforce. |  | 
+| Token Endpoint a| String |  URL pointing to the server responsible for providing the authentication token. According to Salesforce it should be `+https://login.salesforce.com/services/oauth2/token+`, or, if implementing for a community, `+https://acme.force.com/customers/services/oauth2/token+` (where `acme.force.com/customers` is your community URL). |  `+https://login.salesforce.com/services/oauth2/token+` | 
+| Reconnection a| <<Reconnection>> |  When an app is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  | 
+| Host a| String |  Hostname of the proxy. If this property is not set, then no proxy is used, otherwise a proxy is used, but a proxy host must be specified. |  |
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Hostname of the proxy. If set, then a proxy host must be specified. If not set, no proxy is used.

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 2]
```diff
@@ -1,889 +1,1187 @@
...
+
+==== Parameters
+
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Request Body a| Object |  The request body |  `#[payload]` | 
+| Target Variable a| String |  The name of a variable to store the operation's output. |  | 
+| Target Value a| String |  An expression to evaluate against the operation's output and to store the expression outcome in the target variable. |  `#[payload]` |
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Please change all descriptions for Target Value to this:

A variable that stores the output of an expression.

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 3]
```diff
@@ -1,889 +1,1187 @@
...
-| Configuration | String | The name of the configuration to use. | |x
-| SObject Root Type a| String |The sObject type of the root of the trees. |  |x
-| SObject Tree a| Array of Object |A list of maps representing the list of trees that need to be created. |  `#[payload]` |
-| Target Variable a| String |  The variable that stores the operation's output. |  |
-| Target Value a| String |  Evaluate this expression against the operation's output and store the results in the target variable. |  `#[payload]` |
-| Reconnection Strategy a| * reconnect
-* reconnect-forever |  A retry strategy in case of connectivity errors. |  |
+| Configuration | String | The name of the configuration to use. | | x
+| SObject Root Type a| String |  SObject type for the root of the trees. |  | x
+| SObject Tree a| Array of Object |  A list of maps containing representing the list of trees to create. |  `#[payload]` |
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Remove either "containing" or "representing"

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 4]
```diff
@@ -1,889 +1,1187 @@
...
-* SALESFORCE-COMPOSITE:VALIDATION
 
 [[postDelete]]
-== Post Delete
+=== Post Delete
 
 `<salesforce-composite:post-delete>`
 
-This method provides metadata for the result of a delete. It takes as input one record from the result returned by executeCompositeBatch(String, java.util.List) which belongs to delete sub-request generated by preDelete(String, java.util.Map).
+Provide metadata for the result of a delete. This operation takes as input a record from the result returned by `executeCompositeBatch(String, java.util.List)` which deletes a subrequest generated by `preDelete(String, java.util.Map)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Add a comma after `executeCompositeBatch(String, java.util.List)`

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 5]
```diff
@@ -1,889 +1,1187 @@
...
 [[postGetLimits]]
-== Post Get Limits
+=== Post Get Limits
 
 `<salesforce-composite:post-get-limits>`
 
-This method provides metadata for the result of a create. It takes as input one record from the result returned by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#executeCompositeBatch(String, java.util.List) which belongs to create sub-request generated by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#preCreate(String, java.util.Map).
 
-=== Parameters
+Provide metadata for the result of create. This operation takes as input one record from the result returned by `executeCompositeBatch(String, java.util.List)` which belongs to create subrequest generated by `preCreate(String, java.util.Map)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Add a comma after `executeCompositeBatch(String, java.util.List)`

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 6]
```diff
@@ -1,889 +1,1187 @@
...
 
 [[postQuery]]
-== Post Query
+=== Post Query
 
 `<salesforce-composite:post-query>`
 
 
-This method provides metadata for the result of a query. It takes as input one record from the result returned by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#executeCompositeBatch(String, java.util.List) which belongs to query sub-request generated by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#preQuery(String, java.util.Map).
+Provide metadata for the result of query. This operation takes as input a record from the result returned by `executeCompositeBatch(String, java.util.List)` which queries a subrequest generated by `preQuery(String, java.util.Map)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Add a comma after `executeCompositeBatch(String, java.util.List)`

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 7]
```diff
@@ -1,889 +1,1187 @@
...
 [[postQueryAll]]
-== Post Query All
-
+=== Post Query All
 `<salesforce-composite:post-query-all>`
 
-This method provides metadata for the result of a create. It takes as input one record from the result returned by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#executeCompositeBatch(String, java.util.List) which belongs to create sub-request generated by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#preCreate(String, java.util.Map).
 
-=== Parameters
+Provide metadata for the result of a create. This operation takes as input a record from the result returned by `executeCompositeBatch(String, java.util.List)` which creates a subrequest generated by `preCreate(String, java.util.Map)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Add a comma after `executeCompositeBatch(String, java.util.List)`

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 8]
```diff
@@ -1,889 +1,1187 @@
...
 
 [[postRetrieve]]
-== Post Retrieve
-
+=== Post Retrieve
 `<salesforce-composite:post-retrieve>`
 
 
-This method provides metadata for the result of a retrieve. It takes as input one record from the result returned by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#executeCompositeBatch(String, java.util.List) which belongs to retrieve sub-request generated by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#preRetrieve(String, java.util.Map).
+Provide metadata for the result of retrieve. This operation takes as input one record from the result returned by `executeCompositeBatch(String, java.util.List)` which belongs to retrieve subrequest generated by preRetrieve(String, java.util.Map).
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Add a comma after `executeCompositeBatch(String, java.util.List)`

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 9]
```diff
@@ -1,889 +1,1187 @@
...
-[[postUpdate]]
-== Post Update
 
+[[postUpdate]]
+=== Post Update
 `<salesforce-composite:post-update>`
 
 
-This method provides metadata for the result of update. It takes as input one record from the result returned by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#executeCompositeBatch(String, java.util.List) which belongs to update sub-request generated by org.mule.modules.salesforce.composite.SalesforceCompositeConnector#preUpdate(String, java.util.Map).
+Provide metadata for the result of an update. This operation takes as input a record from the result returned by `executeCompositeBatch(String, java.util.List)` which updates a subrequest generated by `preUpdate(String, java.util.Map)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Add a comma after `executeCompositeBatch(String, java.util.List)`

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 10]
```diff
@@ -1,889 +1,1187 @@
...
-[[preDelete]]
-== Pre-Delete
 
+[[preDelete]]
+=== Pre-Delete
 `<salesforce-composite:pre-delete>`
 
 
-This method provides metadata for delete based on object that is going to be deleted and generates a sub-request to be given for execution to executeCompositeBatch(String, java.util.List).
+Provide metadata for a delete based on object to delete and generate a subrequest to be provided for execution by `executeCompositeBatch(String, java.util.List)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: This description is not clear. Is this what it is trying to say?

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 11]
```diff
@@ -1,889 +1,1187 @@
...
 | Name | Type | Description | Default Value | Required
-| Configuration | String | The name of the configuration to use. | |x
-| Type a| String |The type of object to delete. |  |x
-| S Object a| Object |The map with the ID of the object to delete mapped to the ID key. |  `#[payload]` |
-| Target Variable a| String |  The variable that stores the operation's output. |  |
-| Target Value a| String |  Evaluate this expression against the operation's output and store the results in the target variable. |  `#[payload]` |
-| Reconnection Strategy a| * reconnect
-* reconnect-forever |  A retry strategy in case of connectivity errors. |  |
+| Configuration | String | The name of the configuration to use. | | x
+| Type a| String |  Type of the object to delete. |  | x
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Remove "the"

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc [hunk 12]
```diff
@@ -1,889 +1,1187 @@
...
 
-=== Throws
 
-* SALESFORCE-COMPOSITE:CONNECTIVITY
+[[preSearch]]
+=== Pre-Search
+`<salesforce-composite:pre-search>`
+
+
+Provide the user a way to input a SOSL String, and generate a subrequest for execution by `executeCompositeBatch(java.util.List)`.
```
Comment (dhirsch1) [inline modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: Change "a" to "an:

**Review by dhirsch1 — CHANGES_REQUESTED** [review]
All of the Output sections show up with headers only. Is there an easy way to remove the ones with no data? If not, then leave them, because I see it was that way with the older version.

**Comment by dancazacu1** [comment]
@dhirsch1 
This documentation is generated by a maven plugin.
In order to solve your comments I would have to change the javadoc in our connector and regenerate this document. By doing so I will override changes made by @george-hoffer on this PR. 

1. "All of the Output sections show up with headers only. Is there an easy way to remove the ones with no data? If not, then leave them, because I see it was that way with the older version." 

- This is how the plugin works. We cannot do anything about this right now.

2. "Please change all descriptions for Target Value to this:

A variable that stores the output of an expression." 

- Documentation for "Target Value" comes from mule sdk. We cannot change description for this from our connector. We could change this directly into document but this would mean that once the documentation is generated these changes will be lost.

File: modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc (hunk 2 above)
Comment (george-hoffer) [reply modules/ROOT/pages/salesforce/salesforce-composite-connector-reference.adoc]: This fix is incorrect. The correct fix is:
| Target Variable a| String |  A variable that stores the output of the **Target Variable** expression. |  | 
| Target Value a| String |  An expression to evaluate against the operation's output. Stores the output in the **Target Variable** string. |  `#[payload]` | 

Fixed globally.

**Review by george-hoffer — APPROVED** [review]

**Comment by george-hoffer** [comment]
Replaced by https://github.com/mulesoft/docs-connectors/pull/345

**Pull request closed without merging**